views.go

```diff
@@ -63,7 +63,7 @@ func LoginGET(c echo.Context) error {
 
 	if username.String() == "" && verification_key.String() == "" {
 		data.LoginForm = true
-		data.CASLogin = GetCasLoginUrl(service.String())
+		data.CASLogin = GetCasLoginUrl(c.Request(), service.String())
```

Comment:
In that case, making this a relative URL would be the most correct.
If fakecas is behind a load balancer or a local dns entry, the form would still submit correctly.